# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium-shim/patches/with_update/revert_remove_AddTargetPeer.patch
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 2
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium-shim/patches/with_update/revert_remove_AddTargetPeer.patch
--- a/security/sandbox/chromium-shim/patches/with_update/revert_remove_AddTargetPeer.patch	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium-shim/patches/with_update/revert_remove_AddTargetPeer.patch	Wed Jun 12 11:10:48 2019 +0100
@@ -114,7 +114,7 @@
        NOTREACHED();
      }
    }
-@@ -460,26 +495,71 @@ ResultCode BrokerServicesBase::SpawnTarg
+@@ -460,25 +495,70 @@ ResultCode BrokerServicesBase::SpawnTarg
      // TODO(wfh): Find a way to make this have the correct lifetime.
      policy_base->AddRef();
  
@@ -135,7 +135,6 @@
    return result;
  }
  
- 
  ResultCode BrokerServicesBase::WaitForAllTargets() {
    ::WaitForSingleObject(no_targets_.Get(), INFINITE);
    return SBOX_ALL_OK;
@@ -285,11 +284,11 @@
 +  //   If the return is ERROR_GENERIC, you can call ::GetLastError() to get
 +  //   more information.
 +  virtual ResultCode AddTargetPeer(HANDLE peer_process) = 0;
+ 
+  protected:
+   ~BrokerServices() {}
  };
  
  // TargetServices models the current process from the perspective
  // of a target process. To obtain a pointer to it use
  // Sandbox::GetTargetServices(). Note that this call returns a non-null
- // pointer only if this process is in fact a target. A process is a target
- // only if the process was spawned by a call to BrokerServices::SpawnTarget().
- //